src/pip/_internal/utils/misc.py

```diff
@@ -687,6 +706,12 @@ def untar_file(filename, location):
                 # https://github.com/python/mypy/issues/1174
                 fn = split_leading_dir(fn)[1]  # type: ignore
             path = os.path.join(location, fn)
+            if not is_within_directory(location, path):
+                raise InstallationError(
+                    'The tar file (%s) has a file (%s) trying to install '
```

Comment:
I would use the `format()` method on the string for new code instead of `%`.

Reply:
I second this suggestion, though this isn't a blocker for this PR.